frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSyncToParentByRebaseAction.java

```diff
@@ -262,9 +263,21 @@ private GitRebaseParams getIdeaRebaseParamsOf(GitRepository repository, IGitReba
     String newBaseBranchFullName = gitRebaseParameters.getNewBaseBranch().getFullName();
     String forkPointCommitHash = gitRebaseParameters.getForkPointCommit().getHash();
 
-    val options = kotlin.collections.SetsKt.hashSetOf(GitRebaseOption.INTERACTIVE, GitRebaseOption.KEEP_EMPTY);
+    val options = kotlin.collections.SetsKt.hashSetOf(GitRebaseOption.INTERACTIVE);
+
+    isGitRebaseOptionEntryAvailable("--empty=drop", gitVersion).map(options::add);
+
     return new GitRebaseParams(gitVersion, currentBranchName, newBaseBranchFullName,
         /* upstream */ forkPointCommitHash, /* selectedOptions */ options, GitRebaseParams.AutoSquashOption.DEFAULT,
         /* editorHandler */ null);
   }
+
+  @UIThreadUnsafe
```

Comment:
Pls create a separate issue to just switch to using `GitRebaseOption.EMPTY_KEEP` once it's available in git4idea, whenever that happens + mention in a TODO here

Reply:
You mean once it's available in git4idea in all intelliJ versions supported by us? Only then can we switch to simply using `GitRebaseOption.EMPTY_KEEP`.

Reply:
Strictly speaking, once the **earliest major of IntelliJ supported by us** already has this options in git4idea (which might happen even as late as 2025)